### PR TITLE
Upgrade to qooxdoo v7

### PR DESCRIPTION
**Pull request by peuter**
Upgrade to latest qx version and "es6ify" the source code by using qooxdoos builtin job for that.
This PR also includes a change to use the qx-builtin tangible theme to get rid of some old (and outdated/unmaintained) dependencies. The theme is used in the manager, and because the tangible theme is based on color calculations it is not possible to get the exact same colors as we have used before. but now the contrast is higher so its even better.

You need to run an `npm install` before testing the changes.

### CODE REVIEW

**Comment by peuter** [comment]
Actually this PR needs a new qooxdoo release which contains https://github.com/qooxdoo/qooxdoo/pull/10453, which is not available yet.

**Comment by ChristianMayer** [comment]
The es6ify should support staying with single quotes (-> https://prettier.io/docs/en/options.html#quotes ).
When it's possible to stay with the single quotes I'd prefer that (tradition, less changes lines, ...).

But (as stated before) I wouldn't make a religion out of that. So when there are hard facts to switch to double quotes I wouldn't put a veto here.

**Comment by peuter** [comment]
> The es6ify should support staying with single quotes (-> https://prettier.io/docs/en/options.html#quotes ). When it's possible to stay with the single quotes I'd prefer that (tradition, less changes lines, ...).

If it has been possible it would have done it. But there are no configuration options for that job. So you have to live with the pre-defined settings that qooxdoo uses in their sources, and they use double quotes.
 
> But (as stated before) I wouldn't make a religion out of that. So when there are hard facts to switch to double quotes I wouldn't put a veto here.

There are no hard facts just the avoidance of putting too much effort in finding a way to get that job to use a custom config. I think we can live with that change as its just a convention we can get used too.

**Comment by peuter** [comment]
Having a look again at the es6ify documention this is more complete now that when I looked at it the first time. So if the prettier-config option is working I will give it a shot.

**Comment by peuter** [comment]
I changed my mind on waiting for a new qooxdoo release before merging this one. It is just a tiny error somewhere in the manager that will be introduced with this change (until qx fix is released). 

But this pull request is a show-stopper for further development because of merge-conflicts later. So I vote for merging this ASAP.

**Comment by ChristianMayer** [comment]
OK, then I'll priorize it

File: client/source/class/cv/io/Client.js
Comment (ChristianMayer) [inline client/source/class/cv/io/Client.js:503]: Can we avoid this type of change?

I think
```
            Object.getOwnPropertyNames(options.listeners).forEach(function (eventName) {
``` 
is far more readable than
```
            Object.getOwnPropertyNames(options.listeners).forEach(function (
              eventName
            ) {
```

File: client/source/class/cv/io/transport/LongPolling.js
Comment (ChristianMayer) [inline client/source/class/cv/io/transport/LongPolling.js:329]: Would look better by using the `-Syntax and stay in one line, i.e.
```
  this.error(`Error! Type: "${str}" ExceptionObject ...`);
```

File: client/source/class/cv/io/transport/Sse.js
Comment (ChristianMayer) [inline client/source/class/cv/io/transport/Sse.js:68]: Also: single line and backtick string would be preferred

File: client/source/class/cv/io/transport/Sse.js
Comment (ChristianMayer) [inline client/source/class/cv/io/transport/Sse.js:163]: Another example where I think the older, compatcer style was more easy to read and understand

File: client/source/class/cv/io/Client.js
Comment (peuter) [reply client/source/class/cv/io/Client.js]: All I can do is changing the allowed line width (default 80) to e.g. 120 to avoid some line breaks. Other options to control this kind of formatting do not seem to be possible, at least I have found nothing in the prettier documentation.

Prettier calls iself an "opinionated code formatter" which implies that you don't have a lot of options to configure it for your own needs.

File: client/source/class/cv/io/transport/LongPolling.js
Comment (peuter) [reply client/source/class/cv/io/transport/LongPolling.js:272]: I will not change every concatenated string to use backticks, thats not the purpose of this pull request. Please note that it still would add linebreaks if the line exceeds the allowed width, even with backticks. All I can do is to increase the allowed line width, to avoid some of the line breaks.

File: source/class/cv/ConfigCache.js
Comment (ChristianMayer) [inline source/class/cv/ConfigCache.js:126]: Does that work - or does that comment need to be in the line above?

File: client/source/class/cv/io/transport/Sse.js
Comment (peuter) [reply client/source/class/cv/io/transport/Sse.js:145]: This case is indeed an opinion that prettier and I don't share. And also it is not possible to avoid this change.

File: source/class/cv/IconConfig.js
Comment (ChristianMayer) [inline source/class/cv/IconConfig.js:12341]: Probably we should treat this as a special case manually keep or upgrade the old file.

File: source/class/cv/ConfigCache.js
Comment (peuter) [reply source/class/cv/ConfigCache.js:126]: As we do not use jshint anymore (replaced by eslint) this comment is obsolete anyways.

File: source/class/cv/parser/pure/widgets/ColorChooser.js
Comment (ChristianMayer) [inline source/class/cv/parser/pure/widgets/ColorChooser.js:84]: Is there an option to keep the `=` aligned?

File: source/class/cv/plugins/Gauge.js
Comment (ChristianMayer) [inline source/class/cv/plugins/Gauge.js:74]: Also here, keeping the `:` alignment would be nice

File: source/class/cv/parser/pure/widgets/ColorChooser.js
Comment (peuter) [reply source/class/cv/parser/pure/widgets/ColorChooser.js:84]: No

Comment (peuter) [reply source/class/cv/parser/pure/widgets/ColorChooser.js:84]: These are all available options: https://prettier.io/docs/en/options.html

**Comment by ChristianMayer** [comment]
Files are looking fine now. This evening I'll then run the test.

**Comment by peuter** [comment]
qooxdoo has release v7.3, the errors are gone now.

**Review by ChristianMayer — APPROVED** [review]

**Pull request merged**
